apis/stacks/v1alpha1/types_test.go

```diff
@@ -28,35 +28,79 @@ var (
 	_ StackInstaller = &ClusterStackInstall{}
 )
 
-func TestStackInstallSpec_Image(t *testing.T) {
+func TestStackInstallSpec_ImageWithSource(t *testing.T) {
+	type want struct {
+		url string
+	}
+
 	tests := []struct {
-		name string
-		spec StackInstallSpec
-		want string
+		name   string
+		reason string
+		spec   StackInstallSpec
+		want   want
 	}{
 		{
-			name: "NoPackageSource",
+			name:   "NoPackageSource",
+			reason: "Packages without StackInstall source should be unchanged",
```

Comment:
i love this "reason" field for each test case that @negz suggested.  it makes the purpose/intent of the test case so much more clear without having to encode it into the test name.